cluster/saltbase/salt/kube-master-addons/init.sls

```diff
@@ -37,10 +37,19 @@ master-docker-image-tags:
   file.touch:
     - name: /srv/pillar/docker-images.sls
 
+# Current containervm image by default has both docker and kubelet
+# running. But during cluster creation stage, docker and kubelet
+# could be overwritten completely, or restarted due flag changes.
+# The ordering of salt states for service docker, kubelet and
+# master-addon below is very important to avoid the race between
+# salt restart docker or kubelet and kubelet start master components.
 kube-master-addons:
   service.running:
     - enable: True
     - restart: True
+    - require:
+      - service: docker
+      - service: kubelet
```

Comment:
Can you explain _why_ the order here is important. My best guess is that because this script is what side-loads the docker images for the master components, so the kubelet won't be able to start them before the images are side-loaded (attempts to fetch them from gcr.io will fail). But I'm not sure how that eliminates the race condition. Won't the images be side loaded right after the kubelet starts (potentially before it restarts docker) -- meaning that there will be a race between when salt decides to side load the images into docker and when the kubelet tries to launch pods from the manifest?

If the kubelet waits to launch pods from the manifest until after the networking is configured, I don't see how this would have any additional effect -- it shouldn't matter when the docker images are side loaded if the kubelet isn't trying to use them until it is safe to do so.